augur/merge.py

```diff
@@ -503,3 +579,85 @@ def quote(s):
     # …and instead quote a final empty string down here if we're still empty
     # after joining all our parts together.
     return quoted if quoted else shquote('')
+
+
+def run_command(command: str, print_stdout: bool):
+    """Run a command, clearly marking any messages that resemble SeqKit output."""
+    process = subprocess.Popen(
+        ['bash', '-euo', 'pipefail', '-c', command],
+        stdout=subprocess.PIPE if print_stdout else subprocess.DEVNULL,
+        stderr=subprocess.PIPE,
+        text=True,
+    )
+
+    if print_stdout:
+        for line in iter(process.stdout.readline, ''):
+            print(line.rstrip())
+
+    for line in iter(process.stderr.readline, ''):
+        # Remove ANSI escape sequences common in SeqKit output.
+        line = re.sub(r'\x1b\[[\d;]+m', '', line)
+
+        # Detect messages from seqkit and append a prefix.
+        if line.startswith("[INFO]"):
+            print_debug(f"[SeqKit] {line.rstrip()}")
+        elif line.startswith("[ERRO]"):
+            print_info(f"[SeqKit] {line.rstrip()}")
+        else:
+            print_info(line.rstrip())
+
+    # Wait for the seqkit process to finish.
+    # FIXME: address potential deadlock issue?
+    # <https://docs.python.org/3/library/subprocess.html#subprocess.Popen.wait>
+    # I think this is safe since process.stdout and process.stderr are iterated manually.
+    return process.wait()
```

Comment:
I don't believe it is safe from deadlock since (when `print_stdout`) all of stdout is read before all of stderr. For example, if the stderr buffer fills up and writers to it block before stdout hits EOF, then I believe we'll block on reading stdout.

Reply:
Thanks for that example. This is no longer applicable with the switch to `run_shell_command` in https://github.com/nextstrain/augur/pull/1631#discussion_r2008282952